### PR TITLE
DDP-7090 - implement in Pepper backend a mechanism to handle JSON translations

### PR DESCRIPTION
**Pull request by evgeniipr**
## Context

**Draft PR.**
**(Note: in order to complete this feature it needs to merge PR https://github.com/broadinstitute/ddp-study-server/pull/1034).**

"Implement in Pepper backend a mechanism to handle translations stored as JSON doc in a new table `i18n_translation`"

These translations to be exported to `i18n_translation` table during a study building (with StudyBuilder) from i18n-files.
The basic idea of the JSON translations support in a backend:
as soon as any template needs to be rendered the backend tries to find it's variables (defined in a template text) in a special cache storing JSON translations: if it is found then template with resolved variables is evaluated and the result is returned. if it is not found then the old approach to evaluate translation is used.
The basic classes which are implemented:
- `I18nTemplateRenderFacade` - main class called to evaluate (or `render`) template.
- `I18nTemplateFromJSONRenderer` - template renderer which gets template variables from JSON doc.
- `I18nTemplateDefaultRenderer` - template default renderer.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiI18nTranslation.sql.stg [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+getI18nTranslations() ::= <<
+select
+        t.i18n_translation_id,
+        t.i18n_translation_doc,
+        t.study_id,
+        t.language_code_id,
+        t.created_at,
+        t.updated_at
+from
+    i18n_translation t
```
Comment (ssettipalli) [inline pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiI18nTranslation.sql.stg:12]: I believe i18n_translation is a new table.
Not seeing the Liquibase script/schema change to create this table !

Comment (ssettipalli) [reply pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiI18nTranslation.sql.stg:12]: ok. I see it as a separate PR https://github.com/broadinstitute/ddp-study-server/pull/1048/files. 
May be that schema change files should be added to this PR

Comment (evgeniipr) [reply pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiI18nTranslation.sql.stg:12]: Yes, I created Liquibase script in a separate PR. May be I should copy the script to this PR and close PR 1048..

Comment (evgeniipr) [reply pepper-apis/src/main/resources/org/broadinstitute/ddp/db/dao/JdbiI18nTranslation.sql.stg:12]: @ssettipalli I added the Liquibase script creating table `i18n_translation` to this PR. And PR https://github.com/broadinstitute/ddp-study-server/pull/1048 is closed.

**Review by MocanaAtBroad — APPROVED** [review]
I am approving as I believe that everything still works after including this code, but I strongly suspect we will have to make more changes. https://broadinstitute.atlassian.net/browse/DDP-7386 will help us validate what we need to improve upon.

**Comment by Trimax** [comment]
> 

Your  suggestion is to implement the changes in terms of this PR, right?

**Comment by MocanaAtBroad** [comment]
I think it is OK to merge.

**Pull request merged**